datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -362,11 +369,26 @@ def _job_loop(self):
     def _set_rate_limit(self, rate_limit):
         if self._rate_limiter.rate_limit_s != rate_limit:
             self._rate_limiter = ConstantRateLimiter(rate_limit)
+    
+    def _run_sync_job_rate_limited(self):
+        if self._rate_limiter.shall_execute():
+            try:
+                self._run_job_traced()
```

Comment:
if the job failed, assume unexpectedly because expected failure should not bubble up to here, do we want to re-execute in next iteration?

Reply:
May be it depends on the job. If job is super heavy and starts to fail in the middle all the time cause of unexpected problems then it's risky to allow it to run more often then it's period.

If on the other hand it was some random unexpected error that is not repetitive then it's better to allow to try again.

I would say it's a safer approach to rate limit regardless of the outcome. But I am ready to discuss further, like I see arguments for both.

Reply:
I think it's better to leave to the job implementation to decide what exception is expected and what is not. The rate limiter always tries to re-run a failed job in check run. It's up to the job implementation to decide whether to wait for the full collection interval to expire (swallow the exception) or not (bubble up the exception). This would be more flexible in case we do want an immediate retry. 
If we always set the last_event timestamp in `finally`, we lose the ability to retry immediately.